tests/src/Unit/SetRolesTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\Tests\og\Unit;
+
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Entity\OgRole;
+use Drupal\og\OgRoleInterface;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests set role method from the og membership.
+ *
+ * @group og
+ * @coversDefaultClass \Drupal\og\Entity\OgMembership
+ */
+class SetRolesTest extends UnitTestCase {
+
+  /**
+   * OG roles service.
+   *
+   * @var \Drupal\og\Entity\OgRole|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $ogRole;
+
+  /**
+   * OG membership service.
+   *
+   * @var \Drupal\og\Entity\OgMembership|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $ogMembership;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    $this->ogRole = $this->prophesize(OgRole::class);
+    $this->ogMembership = $this->prophesize(OgMembership::class);
```

Comment:
actually remove this one - you shouldn't `prophesize` the class you are checking, as it misses the whole point ;)